### PR TITLE
(PC-30943) fix(Offer): Fix duration format on offer

### PR DESCRIPTION
**Pull request by inesmouandjolobe**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-30943

## Flakiness

If I had to re-run tests in the CI due to flakiness, I add the incident [on Notion][2]

## Checklist

I have:

- [X] Made sure my feature is working on web.
- [X] Made sure my feature is working on mobile (depending on relevance : real or virtual devices)
- [X] Written **unit tests** native (and web when implementation is different) for my feature.
- [X] Added a **screenshot** for UI tickets or deleted the screenshot section if no UI change
- [X] If my PR is a bugfix, I add the link of the "résolution de problème sur le bug" [on Notion][1]

## Screenshots

**delete** _if no UI change_

| Platform         | Mockup/Before | After |
| :--------------- | :-----------: | :---: |
| Desktop - Chrome | Capture d’écran 2024-08-12 à 12 06 44 | Capture d’écran 2024-08-12 à 12 07 20 |

[1]: https://www.notion.so/passcultureapp/R-solution-de-probl-mes-sur-les-bugs-5dd6df8f6a754e6887066cf613467d0a
[2]: https://www.notion.so/passcultureapp/cb45383351b44723a6f2d9e1481ad6bb?v=10fe47258701423985aa7d25bb04cfee&pvs=4

### CODE REVIEW

File: src/features/offer/helpers/formatDuration/formatDuration.ts [hunk 1]
```diff
@@ -6,7 +6,10 @@ export const formatDuration = (duration: number | null | undefined): string => {
 
   if (minutes === 0) {
     return `${hours}h`
+  } else if (minutes < 10) {
```
Comment (cedricls1) [inline src/features/offer/helpers/formatDuration/formatDuration.ts:9]: On peut simplifier la fonction :

```
export const formatDuration = (duration: number | null | undefined): string => {
  if (!duration) return '-';

  const hours = Math.floor(duration / 60);
  const minutes = duration % 60;

  if (hours === 0) {
    return `${minutes} minutes`;
  }

  const formattedMinutes = minutes < 10 ? `0${minutes}` : minutes;
  return `${hours}h${formattedMinutes}`;
};
```

Comment (inesmouandjolobe) [reply src/features/offer/helpers/formatDuration/formatDuration.ts:9]: Comment tu gères le cas de l'heure sans la minute ? (ex : `2h`)

Comment (cedricls1) [reply src/features/offer/helpers/formatDuration/formatDuration.ts:9]: Ah yes bien vu :
```
export const formatDuration = (duration: number | null | undefined): string => {
  if (!duration) return '-'

  const hours = Math.floor(duration / 60)
  const minutes = duration % 60

  if (hours === 0) {
    return `${minutes} minutes`
  }

  if (minutes === 0) {
    return `${hours}h`;
  }

  const formattedMinutes = minutes < 10 ? `0${minutes}` : minutes
  return `${hours}h${formattedMinutes}`
}
```

Comment (inesmouandjolobe) [reply src/features/offer/helpers/formatDuration/formatDuration.ts]: C'est push :)

**Review by cedricls1 — APPROVED** [review]

**Pull request merged**